src/consumption.cpp

```diff
@@ -732,6 +732,19 @@ void player::consume_effects( item &food, bool rotten )
             fun = 1;
         }
     }
+    //Food is made out of meat (use ALLERGEN_MEAT, not CARNIVORE_OK (the latter has eggs and milk),
+    //Reduces bad morale penalties on food only made of meat.
+    if( has_trait( "CARNIVORE" ) && food.has_flag( "ALLERGEN_MEAT" ) &&
+        !food.has_any_flag( carnivore_blacklist ) &&
+        !( food.has_flag( "CANNIBALISM" ) && !has_trait( "CANNIBAL" ) ) ) {
```

Comment:
Will look at it, I do recall that sapiovore had carnivore as a prereq, and not sure if psycho should give bonusses, but as carnivore is a neg trait, and psycho is something that costs points, I think for game balance, it woul be allright. And I'll change the fun line as well.